### PR TITLE
setup: categories and custom kwargs in ep loading

### PR DESCRIPTION
**Pull request by krzysztof**
* Adds categories and other customization kwargs to admin entry point.

Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

**Comment by krzysztof** [comment]
@lnielsen @jirikuncar 

This patch addresses #4 
Perhaps something else should be done before people start writing admin interfaces for their modules ?

File: invenio_admin/ext.py [hunk 1]
```diff
@@ -62,12 +62,20 @@ def init_app(self, app, entry_point_group='invenio_admin.views',
...
             for ep in pkg_resources.iter_entry_points(group=entry_point_group):
-                modelview, model = ep.load()
+                adminview_dict = ep.load()
+                assert 'model' in adminview_dict, \
+                    "Admin's entrypoint dictionary must define the 'model'"
+                assert 'modelview' in adminview_dict, \
+                    "Admin's entrypoint dictionary must define the 'modelview'"
+                model = adminview_dict['model']
+                modelview = adminview_dict['modelview']
+                kwargs = dict((k, v) for k, v in adminview_dict.items()
```
Comment (lnielsen) [inline invenio_admin/ext.py]: You can use dict comprehension here instead:
```python
{k: v for (k,v) in d.items() if k not in ('model', 'modelview')}
```

File: invenio_admin/ext.py [hunk 2]
```diff
@@ -62,12 +62,20 @@ def init_app(self, app, entry_point_group='invenio_admin.views',
         # Load administration interfaces defined by entry points.
         if entry_point_group:
             for ep in pkg_resources.iter_entry_points(group=entry_point_group):
-                modelview, model = ep.load()
+                adminview_dict = ep.load()
+                assert 'model' in adminview_dict, \
+                    "Admin's entrypoint dictionary must define the 'model'"
+                assert 'modelview' in adminview_dict, \
+                    "Admin's entrypoint dictionary must define the 'modelview'"
+                model = adminview_dict['model']
```
Comment (jirikuncar) [inline invenio_admin/ext.py]:
```python
model = adminview_dict.pop('model')
modelview = adminview_dict.pop('modelview')
ProtectedAdminView = protected_adminview_factory(modelview)
self.admin.add_view(
    ProtectedAdminView(model, db.session, **adminview_dict)
)
```

Comment (krzysztof) [inline invenio_admin/ext.py]: That will modify the dictionary on the other end, I believe making a copy is safer.

Comment (jirikuncar) [inline invenio_admin/ext.py]:
```python
adminview_dict = dict(ep.load())
```

Comment (krzysztof) [inline invenio_admin/ext.py]: What's the advantage of doing a ``pop`` at this point then ?

Comment (krzysztof) [inline invenio_admin/ext.py]:
```
python -m timeit "d = {k: k for k in range(5)}; d2 = dict(d); d2.pop(0); d2.pop(1)"
1000000 loops, best of 3: 0.949 usec per loop
```
```
python -m timeit "d = {k: k for k in range(5)}; d2 = {k: v for (k,v) in d.items() if k not in (0, 1)}"
1000000 loops, best of 3: 1.22 usec per loop
```

Well.. its 30% faster :)

Comment (jirikuncar) [inline invenio_admin/ext.py]: only 30%? you have really small dict :stuck_out_tongue_winking_eye:

Comment (krzysztof) [inline invenio_admin/ext.py]: For just two items the other one is faster though ;)

**Comment by krzysztof** [comment]
Done writing up the docstring, comments are welcomed.

File: invenio_admin/__init__.py [hunk 3]
```diff
@@ -21,8 +21,139 @@
...
+>>> with app.app_context():
+...     create_database(db.engine.url)
+...     db.create_all()
+>>> app.run() # doctest: +SKIP
+
+
+You should now be able to access the admin panel `http://localhost:5000/admin
+<http://localhost:5000/admin>`_.
+
+Build-In security and authentication check
```
Comment (lnielsen) [inline invenio_admin/__init__.py]: Built-in

File: invenio_admin/__init__.py [hunk 4]
```diff
@@ -21,8 +21,139 @@
...
+--------------
+This section serves as an example on basic usage of the Invenio-Admin.
+Create a basic Flask application (Flask-CLI is not needed for Flask 1.0 and up)
+
+>>> from flask import Flask
+>>> from flask_cli import FlaskCLI
+>>> app = Flask('DinerApp')
+>>> ext_cli = FlaskCLI(app)
+
+InvenioDB is the only required Invenio dependency which needs to be initiated:
```
Comment (lnielsen) [inline invenio_admin/__init__.py]: initiated -> initialized ?

File: invenio_admin/__init__.py [hunk 5]
```diff
@@ -21,8 +21,139 @@
...
+Build-In security and authentication check
+------------------------------------------
+Although Invenio-Admin does not directly depend on Invenio-Access or
+Invenio-Accounts module, it does protect the admin views with Flask-Login and
+Flask-Principal features. In order to log in to a Invenio-Admin panel the user
+needs to be authenticated using Flask-Login.
+
+AdminView discovery through setuptools' entry points
+----------------------------------------------------
+Default way of adding model views to admin panel in Invenio is though the
```
Comment (lnielsen) [inline invenio_admin/__init__.py]: The default

File: setup.py [hunk 6]
```diff
@@ -62,7 +62,7 @@
 
 install_requires = [
     'Flask-Admin>=1.3.0',
-    'Flask-Login<0.3.0,>=0.2.11',
+    'Flask-Login>=0.3.2',
```
Comment (lnielsen) [inline setup.py]: Flask-Security has not yet been released with support for Flask-Login 0.3.0+ so we need to keep the ``'Flask-Login<0.3.0,>=0.2.11',`` still.

**Comment by lnielsen** [comment]
Looks good. Thanks for writing up the documentation :-)

File: invenio_admin/__init__.py [hunk 7]
```diff
@@ -21,8 +21,139 @@
...
+
+... and register them in the admin extension:
+
+>>> ext_admin.admin.add_view(LunchModelView(Lunch, db.session))
+
+Finally, initialize the database and run the development server:
+
+>>> from sqlalchemy_utils.functions import create_database
+>>> app.config.update(SQLALCHEMY_DATABASE_URI='sqlite:///test.db',
+...     SECRET_KEY='XY')
```
Comment (krzysztof) [inline invenio_admin/__init__.py]: Change XY to something else

File: invenio_admin/ext.py [hunk 8]
```diff
@@ -42,8 +42,9 @@ class InvenioAdmin(object):
         from.
     """
 
-    def __init__(self, app=None, **kwargs):
+    def __init__(self, app=None, anonymous=False, **kwargs):
         """Extension initialization."""
```
Comment (krzysztof) [inline invenio_admin/ext.py]: Add a docstring on anonymous parameter

File: invenio_admin/__init__.py (hunk 4 above)
Comment (krzysztof) [inline invenio_admin/__init__.py]: Actually I wanted to use ``instantiated``

**Comment by krzysztof** [comment]
@lnielsen I have resolved the comments mentioned above.

**Pull request merged**